gql/versions.go

```diff
@@ -124,9 +124,12 @@ func FinalizeVersionUploadAsync(ctx context.Context, mod *postgres.Mod, versionI
 	}
 
 	// TODO Validate mod files
-	success, key := storage.RenameVersion(ctx, mod.ID, mod.Name, versionID, modInfo.Version)
+	//Okay, uploaded file is read and readable... let's dump it and separate, and re-save as ModName-Combined-SemVersion in ModLink
+	storage.DeleteVersion(ctx, mod.ID, mod.Name, versionID)
 
-	if !success {
+	separated, key := storage.SeparateMod(ctx, fileData, mod.ID, mod.Name, dbVersion.ID, modInfo.Version)
```

Comment:
Is this correct? You are deleting the version and then splitting it?

Reply:
Deleting, splitting, then saving it again with a ModArch to send over for validation...

The delete and rename, essentially, is no different from storage.RenameVersion. I can reuse that code if you desire.

Reply:
Also, at that point in the code, the file has been read into memory, I am just sending the buffer/bytes to SeparateMod that I am going to revisit to cut down repeat code.

Reply:
In the `RenameVersion` function it calls to the underlying `Rename` which actually just copies the object, and THEN it deletes it from storage, not the other way around.

Reply:
https://github.com/satisfactorymodding/smr-api/pull/12/commits/2ac1243aed7bd5d4641f015af08a919157fc46a0
https://github.com/satisfactorymodding/smr-api/pull/12/commits/c078aa8776d72806e5fb249eef0a7d77d31c3929

Let me know if this is up to par... ignore cleanPlatform, it was a foolish idea, and I have removed it.

Reply:
https://github.com/satisfactorymodding/smr-api/pull/12/commits/dd24599d4a6af63705883b7b89dc47b956820c32

Let me know if this is more desired.